internal/providers/zvm/zvm.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2019 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// The zVM provider fetches a local configuration from the virtual unit
+// record devices.
+
+package zvm
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os/exec"
+	"strings"
+
+	"github.com/coreos/ignition/v2/config/shared/errors"
+	"github.com/coreos/ignition/v2/config/v3_1_experimental/types"
+	"github.com/coreos/ignition/v2/internal/distro"
+	"github.com/coreos/ignition/v2/internal/log"
+	"github.com/coreos/ignition/v2/internal/providers/util"
+	"github.com/coreos/ignition/v2/internal/resource"
+	"github.com/coreos/vcontext/report"
+)
+
+const readerDevice string = "000c"
+
+func FetchConfig(f *resource.Fetcher) (types.Config, report.Report, error) {
+	// Fetch config files directly from reader device.
+	_, err := f.Logger.LogCmd(exec.Command(distro.ModprobeCmd(), "vmur"), "Loading zVM control program module")
+	if err != nil {
+		f.Logger.Err("Couldn't install vmur module: %v", err)
+		errors := fmt.Errorf("Couldn't install vmur module: %v", err)
```

Comment:
Minor: can probably dedupe these by assigning to a var first?
Edit: and same in other spots below

Reply:
> Minor: can probably dedupe these by assigning to a var first?
> Edit: and same in other spots below

Did you mean a function like printError() to do f.loggr.Err? I don't think printError should replace f.logger.Err to involve the context.

Reply:
I just mean something like:

```go
errmsg := fmt.Sprintf("Couldn't install vmur module: %v", err)
f.Logger.Err(errmsg)
errors := fmt.Errorf(errmsg)
```